src/util-lua-smtp.c

```diff
@@ -0,0 +1,477 @@
+    }
+
+    /* Create a new table in luastate for rcpt list */
+    lua_newtable(luastate);
+    /* rcpt var for iterator */
+    int u = 1;
+    SMTPString *rcpt;
+
+    TAILQ_FOREACH(rcpt, &smtp_tx->rcpt_to_list, next) {
+        LuaPushStringBuffer(luastate, rcpt->str, rcpt->len);
+        lua_pushinteger(luastate, u++);
+        lua_settable(luastate, -3);
+    }
+    /* return 1 since we allways push one table to luastate */
+    return 1;
+}
+
+/**
+ * \brief function loops through rcpt-list located in
+ * flow->SMTPState->SMTPTransaction, adding all items to a table.
+ * Then pushing it to the luastate stack.
+ *
+ * \params luastate luastate stack for internal communication with Lua.
+ * Used to hand over data to the recieveing luascript.
+ *
+ * \retval 1 if the table is pushed to lua.
+ * Returns error int and msg pushed to luastate stack if error occurs
+ */
+
+static int SMTPGetRcptList(lua_State *luastate)
+{
+    /* check protocol */
+    if(!(LuaStateNeedProto(luastate, ALPROTO_SMTP))) {
+        return LuaCallbackError(luastate, "Error: protocol not SMTP");
+    }
+    /* create lockhint var for flowlock check. */
+    int lock_hint = 0;
+    /* Extract flow, with lockhint to check mutexlocking */
+    Flow *flow = LuaStateGetFlow(luastate, &lock_hint);
+    if(flow == NULL) {
+        return LuaCallbackError(luastate, "Internal error: no flow");
+    }
+    /* check if already mutexlocked by parents */
+    if(lock_hint == LUA_FLOW_NOT_LOCKED_BY_PARENT) {
+        /* lock flow */
+        FLOWLOCK_RDLOCK(flow);
+        GetRcptList(luastate, flow);
+        /* open flow */
+        FLOWLOCK_UNLOCK(flow);
+    } else {
+        GetRcptList(luastate, flow);
+    }
+    /* return 1 since we allways push one table to luastate */
+    return 1;
+}
+
+/**
+ * \brief intern function used by SMTPGetAttachmentInfo
+ *
+ * \params luastate, luastate for internal communication towards the
+ * luascripting engine.
+ *
+ * \retval 1 if the table is pushed to lua.
+ * Returns error int and msg pushed to luastate stack if error occurs
+ */
+
+static int GetAttachmentInfo(lua_State *luastate, Flow *flow)
+{
+    /* Extract SMTPState from flow */
+    SMTPState *state = (SMTPState *) FlowGetAppState(flow);
+    if(state == NULL) {
+        return LuaCallbackError(luastate, "error: state not found");
+    }
+    /* get FileContainer in SMTPState */
+    FileContainer *file_con = state->files_ts;
+    if(file_con == NULL) {
+        return LuaCallbackError(luastate, "error: no files found");
+    }
+    /* point to start of list for iterating trough */
+    File *file = file_con->head;
+    if(file == NULL) {
+        return LuaCallbackError(luastate, "error: no file(s) in container");
+    }
+    int u = 1;
+    /* create new table for placement of findings */
+    lua_newtable(luastate);
+    /* loop through and push filename to luastate table on stack */
+    while(file != NULL) {
```

Comment:
files are stored per smtp state, not per 'transaction'. Here simply all files are returned. I think it would make more sense to act only on the files belong to the current tx?